test/odbc/psqlodbc/xml.cpp

```diff
@@ -0,0 +1,230 @@
+    pg_test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", PG_TABLE_NAME));
+  }
+
+  void TearDown() override {
+    if (!Drivers::DriverExists(ServerType::PSQL)) {
+      GTEST_SKIP() << "PSQL Driver not present: skipping all tests for this fixture.";
+    }
+    if (!Drivers::DriverExists(ServerType::MSSQL)) {
+      GTEST_SKIP() << "MSSQL Driver not present: skipping all tests for this fixture.";
+    }
+
+    OdbcHandler bbf_test_setup(Drivers::GetDriver(ServerType::MSSQL));
+    bbf_test_setup.ConnectAndExecQuery(DropObjectStatement("VIEW", BBF_VIEW_NAME));
+    bbf_test_setup.CloseStmt();
+    bbf_test_setup.ExecQuery(DropObjectStatement("TABLE", BBF_TABLE_NAME));
+    
+    OdbcHandler pg_test_setup(Drivers::GetDriver(ServerType::PSQL));
+    pg_test_setup.ConnectAndExecQuery(DropObjectStatement("VIEW", PG_VIEW_NAME));
+    pg_test_setup.CloseStmt();
+    pg_test_setup.ExecQuery(DropObjectStatement("TABLE", PG_TABLE_NAME));
+  }
+};
+
+TEST_F(PSQL_DataTypes_Xml, Table_Creation) {
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  const vector<int> BBF_LENGTH_EXPECTED = {10, 0};
+  const vector<int> BBF_PRECISION_EXPECTED = {10, 0};
+  const vector<int> BBF_SCALE_EXPECTED = {0, 0};
+  const vector<string> BBF_NAME_EXPECTED = {"int", "xml"};
+
+  testCommonColumnAttributes(ServerType::MSSQL, BBF_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    BBF_LENGTH_EXPECTED, BBF_PRECISION_EXPECTED, 
+    BBF_SCALE_EXPECTED, BBF_NAME_EXPECTED);
+
+  const vector<int> PG_LENGTH_EXPECTED = {4, 255};
+  const vector<int> PG_PRECISION_EXPECTED = {0, 0};
+  const vector<int> PG_SCALE_EXPECTED = {0, 0};
+  const vector<string> PG_NAME_EXPECTED = {"int4", "xml"};
+
+  testCommonColumnAttributes(ServerType::PSQL, PG_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    PG_LENGTH_EXPECTED, PG_PRECISION_EXPECTED, 
+    PG_SCALE_EXPECTED, PG_NAME_EXPECTED);
+
+  // Clean-up
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Insertion_Success) {
+  vector<string> inserted_values = {
+    "NULL",
+    SIMPLE_XML,
+    NORMAL_XML,
+    LARGE_XML,
+  };
+  const int NUMBER_OF_ENTRIES = inserted_values.size();
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  insertValuesInTable(ServerType::PSQL, PG_TABLE_NAME, inserted_values, false, NUMBER_OF_ENTRIES);
+
+  inserted_values = duplicateElements(inserted_values);
+
+  verifyValuesInObject(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Data_Length) {
```

Comment:
What's the difference between this test and `Insertion_Success`

Reply:
Excellent question.  I had a different version of this at one point, but it wasn't working so I thought I removed it, but I guess I didn't.

Will remove it.